bindings/java/src/main/java/org/ucx/jucx/NativeLibs.java

```diff
@@ -69,6 +65,41 @@ private static void loadLibrary(String resourceName) {
         }
     }
 
+    /**
+     * Extracts shared UCT transport.
+     */
+    private static void extractUCTLibs() {
+        URL ucxResource = loader.getResource("ucx");
+        File ucxFolder = new File(ucxResource.getPath());
+        Path ucxTempFolder;
+        try {
+            createTempDir();
+            ucxTempFolder = Files.createDirectory(Paths.get(tempDir.getPath(), "ucx"));
+        } catch (IOException ex) {
+            errorMessage = "Failed to create temp directory";
+            return;
+        }
+        for (File uctLib: ucxFolder.listFiles()) {
+            if (!uctLib.getName().startsWith("libuct_")) {
+                continue;
+            }
+            FileOutputStream os = null;
+            FileInputStream is = null;
+            File out = new File(ucxTempFolder.toAbsolutePath().toString(), uctLib.getName());
```

Comment:
why do we need to extract to temp dir?

Reply:
Java distributes binaries as jar files (archives with java byte code and all uct shared libraries). We need to extract from a tarball to temp directory to be able to load them, since you can't do dlsym from a tarball

Reply:
and where is libuct.so extracted to?

Reply:
Java creates temporary directory (configured by `java.io.tmpdir` java property). Then it puts all shared libraries from jucx jar archive there.

Reply:
is this the same tmpdir for all processes on the system?

Reply:
No each process creates it's own tmp directory with jucx prefix and some random number (something like /tmp/jucx1234567). On jvm successful exit this directory is deleted